klp_configure.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Copyright (C) 2019 SUSE
+# Author: Libor Pechacek
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+set -e
+. "$(dirname $0)"/klp_tc_functions.sh
+
+# compile-test for hrtimer API ()
+COMPILETEST_DIR=/tmp/live-patch/klp_compile_test
+mkdir -p $COMPILETEST_DIR
+cp "$SOURCE_DIR"/klp_compile_test_hrtimer.c $COMPILETEST_DIR/
+
+echo -n 'export KLP_TEST_HRTIMER_OLD='
+if klp_compile_module /tmp/live-patch/klp_compile_test/klp_compile_test_hrtimer.c > /dev/null 2>&1;
+then
+	echo "1"
+else
+	echo "0"
+fi
+
+# Check for getpid syscall prefix
+echo -n 'export KLP_TEST_SYSCALL_FN_PREFIX='
+sed '/T [[:alnum:]_]*sys_getpid$/{ s/.* \([^ ]*\)sys_getpid$/\1/;p }; d' /proc/kallsyms | tail -1
```

Comment:
If I'm not mistaken, there are two getpid() syscall stubs on x86_64 (starting with 4.17.0): __x64_sys_getpid() and __ia32_sys_getpid(). Likewise, on s390, there should be __s390x_sys_getpid() and __s390_sys_getpid(). Are you sure the above always catches the right variant? If not, we should implement some logic based on the kernel version and arch, similar to what is done in kgraft-patches/syscalls.h (master-livepatch branch) for the KLP_SYMCALL_SYM define.

Reply:
Good point. I happily took the last match in the hope that it will be the right one. That's overly fragile though. I'll glance at `kgraft-patches/syscalls.h`.

As an alternative, I thought about searching for `__$(uname -m)_sys_getpid` in kallsyms, and if it's not present, resort to plain `sys_getpid` (with possible test abort in case that even this symbol is missing). I'll weight the alternatives after studying your proposal.

Reply:
OK, the search for `__$(uname -m)_sys_getpid` didn't work. The thing is that on `x86_64` the syscall function name is `__x64_sys_getpid`. I'll rewrite the [kgraft-patches checks](https://kernel.suse.com/cgit/kernel-livepatch/tree/klp_syscalls.h?h=SLE15-SP2_Update_0) to shell then.

Reply:
So I've implemented a kernel version comparison and created a switch based on `uname -m`. Hope that's a better implementation.